pkg/metrics/metrics.go

```diff
@@ -12,8 +12,10 @@ var (
 	log                  = logger.NewLogger("metrics")
 )
 
+// InitCollector is the type for functions that initialize collectors.
 type InitCollector func() (prometheus.Collector, error)
 
+// RegisterCollector registers the names prometheus.Collector for metrics collection.
```

Comment:
nit: maybe it's just me but I don't understand the meaning of *the names prometheus.Collector* 😄

Reply:
Argh... should have been *named*. Fixed.